src/Widgets/PreferencesDialog.vala

```diff
@@ -0,0 +1,203 @@
+public class PreferencesDialog : Gtk.Window {
+    private Gtk.FontButton font_btn;
+
+    public PreferencesDialog (Window parent, Preferences prefs) {
```

Comment:
I like to separate the UI setup code from the signal connect code, it would be awesome if you could do that.

constructor should have
setup_ui ();
setup_signals ();

(and try to avoid lambda {but this is  not very important})